Assets/HotmaxEditorVR/loadPropsMenu.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+public class loadPropsMenu : MonoBehaviour
+{
+    //--local refs
+    private stateManager _stateManagerMutatorRef;
+    private stateManager.editorModes _editorMode = stateManager.editorMode;
+    private bool _playerIsLocomoting = stateManager.playerIsLocomoting;
+
+    //--child gameObject refs
+    public GameObject propTiles;
+    
+    //--Lists to handle getting unique props in menu
+    private List<string> propName = new List<string>();
+    private List<GameObject> propObject = new List<GameObject>();
+    private List<List<GameObject>> chunkedList = new List<List<GameObject>>();
+
+    //--if false, disabled initial props on stage
+    //--TODO: remove when props are received from PropClass
+    public bool propIsInStageAlready = false;
+
+    private int currentSpawnPage;
+
+    void Start()
+    {
+        _stateManagerMutatorRef = GameObject.FindObjectOfType(typeof(stateManager)) as stateManager;
+
+        //get the names of all props in props
+        //TODO: we will be getting the props from a propClass in the form of a List of Serialized GameObjects
+        for (int i = 0; i < init.props.transform.childCount; i++)
+        {
+            propName.Add(init.props.transform.GetChild(i).gameObject.name);
+        }
+
+        //weed out props that have the same name
+        propName = propName.Distinct().ToList();
+
+        //copy the game objects with the unique names to a gameObject list 
+        for (int i = 0; i < propName.Count; i++)
+        {
+            propObject.Add(GameObject.Find(propName[i]));
+        }
+
+        //set all game objects in props inactive if they are not in the stage
+        if (!propIsInStageAlready)
+        {
+            for (int i = 0; i < init.props.transform.childCount; i++)
+            {
+                init.props.transform.GetChild(i).gameObject.SetActive(false);
+            }
+        }
+
+        //create "chunks" groups of 4 to display the object
+        chunkedList = ListExtensions.ChunkBy(propObject, 4);
+        showPropObjectPage(0);
+    }
+
+   public void spawnPageUp()
+    {
+        if (currentSpawnPage >= chunkedList.Count - 1)
+        {
+            return;
+        }
+        else
+        {
+            currentSpawnPage += 1;
+            showPropObjectPage(currentSpawnPage);
+        }
+    }
+
+    public void spawnPageDown()
+    {
+        if (currentSpawnPage <= 0)
+        {
+            return;
+        }
+        else
+        {
+            currentSpawnPage -= 1;
+            showPropObjectPage(currentSpawnPage);
+        }
+    }
+
+    void showPropObjectPage(int page)
+    {
+        currentSpawnPage = page;
+        //remove old tiles before loading new ones
+        for (int i = 0; i < propTiles.transform.childCount; i++)
+        {
+            Destroy(propTiles.transform.GetChild(i).gameObject);
```

Comment:
i would just deactivate instead of destroy - since you've already done the work of instantiating them.  then on page check if they already exist, if they don't then instantiate - if they do then just active them

Reply:
what do you mean by "then on page"?

Reply:
on spawnPage change

Reply:
leaving it alone for now